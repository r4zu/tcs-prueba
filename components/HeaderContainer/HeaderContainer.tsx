import { Text, View, StyleSheet, Platform } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { FontAwesome6 } from '@expo/vector-icons';

interface HeaderContainerProps {
  children: React.ReactNode;
}

export const HeaderContainer: React.FC<HeaderContainerProps> = ({
  children,
}) => {
  return (
    <SafeAreaView style={styles.container} testID="safe-area-view">
      <View style={styles.content} testID="content-view">
        <View style={styles.header} testID="header-view">
          <View testID="font-awesome-icon">
            <FontAwesome6 name="money-bills" size={24} color="#182F63" />
          </View>
          <Text style={styles.text} testID="header-text">
            BANCO
          </Text>
        </View>
        <View style={styles.separator} testID="separator-view" />
        {children}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
  },
  content: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginTop: Platform.OS === 'android' ? 10 : 0,
    paddingTop: Platform.OS === 'ios' ? 10 : 0,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 10,
  },
  text: {
    marginLeft: 10,
    fontSize: 16,
    fontWeight: 'bold',
    color: '#182F63',
  },
  separator: {
    width: '100%',
    height: 1,
    backgroundColor: '#ccc',
  },
});
