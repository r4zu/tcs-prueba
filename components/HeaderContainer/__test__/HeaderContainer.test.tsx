import React from 'react';
import { Text, Platform } from 'react-native';
import { render } from '@testing-library/react-native';

import { HeaderContainer } from '../HeaderContainer';

describe('HeaderContainer component', () => {
  it('renders correctly on Android', () => {
    Platform.OS = 'android';
    const { getByText, getByTestId } = render(
      <HeaderContainer>
        <Text>Child Component</Text>
      </HeaderContainer>
    );

    // Check if the SafeAreaView is rendered
    const safeAreaView = getByTestId('safe-area-view');
    expect(safeAreaView).toBeTruthy();

    // Check if the FontAwesome6 icon is rendered
    const icon = getByTestId('font-awesome-icon');
    expect(icon).toBeTruthy();

    // Check if the header text is rendered
    const headerText = getByTestId('header-text');
    expect(headerText).toBeTruthy();

    // Check if the separator is rendered
    const separator = getByTestId('separator-view');
    expect(separator).toBeTruthy();

    // Check if the children are rendered correctly
    const childComponent = getByText('Child Component');
    expect(childComponent).toBeTruthy();
  });

  it('renders correctly on iOS', () => {
    Platform.OS = 'ios';
    const { getByText, getByTestId } = render(
      <HeaderContainer>
        <Text>Child Component</Text>
      </HeaderContainer>
    );

    // Check if the SafeAreaView is rendered
    const safeAreaView = getByTestId('safe-area-view');
    expect(safeAreaView).toBeTruthy();

    // Check if the FontAwesome6 icon is rendered
    const icon = getByTestId('font-awesome-icon');
    expect(icon).toBeTruthy();

    // Check if the header text is rendered
    const headerText = getByTestId('header-text');
    expect(headerText).toBeTruthy();

    // Check if the separator is rendered
    const separator = getByTestId('separator-view');
    expect(separator).toBeTruthy();

    // Check if the children are rendered correctly
    const childComponent = getByText('Child Component');
    expect(childComponent).toBeTruthy();
  });

  it('matches snapshot on Android', () => {
    Platform.OS = 'android';
    const { toJSON } = render(
      <HeaderContainer>
        <Text>Snapshot Test</Text>
      </HeaderContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot on iOS', () => {
    Platform.OS = 'ios';
    const { toJSON } = render(
      <HeaderContainer>
        <Text>Snapshot Test</Text>
      </HeaderContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
