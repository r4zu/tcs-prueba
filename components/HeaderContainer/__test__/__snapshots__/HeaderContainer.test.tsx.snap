// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer component matches snapshot on Android 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
      "width": "100%",
    }
  }
  testID="safe-area-view"
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginTop": 0,
        "paddingTop": 10,
      }
    }
    testID="content-view"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
      testID="header-view"
    >
      <View
        testID="font-awesome-icon"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#182F63",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "FontAwesome6Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#182F63",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
        testID="header-text"
      >
        BANCO
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#ccc",
          "height": 1,
          "width": "100%",
        }
      }
      testID="separator-view"
    />
    <Text>
      Snapshot Test
    </Text>
  </View>
</RNCSafeAreaView>
`;

exports[`HeaderContainer component matches snapshot on iOS 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
      "width": "100%",
    }
  }
  testID="safe-area-view"
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginTop": 0,
        "paddingTop": 10,
      }
    }
    testID="content-view"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
      testID="header-view"
    >
      <View
        testID="font-awesome-icon"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#182F63",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "FontAwesome6Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#182F63",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
        testID="header-text"
      >
        BANCO
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#ccc",
          "height": 1,
          "width": "100%",
        }
      }
      testID="separator-view"
    />
    <Text>
      Snapshot Test
    </Text>
  </View>
</RNCSafeAreaView>
`;
